src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
One way to reduce complexity is to stick to the single responsibility principle. I see several concerns (pagination, selection, sorting, path and URL definition) in this module which could be isolated in their own respective modules, bringing down complexity.